src/test/scala/pureconfig/PureconfSuite.scala

```diff
@@ -446,4 +446,38 @@ class PureconfSuite extends FlatSpec with Matchers with OptionValues with TryVal
     val files = fileList()
     loadConfigFromFiles[FlatConfig](files).failure.exception.getMessage should include regex "config files.*must not be empty"
   }
+
+  it should "load configurations which use += to append into a list from multiple files with somewhat surprising results" in {
+    val files = fileList(
+      "src/test/resources/conf/loadConfigFromFiles/appendIntoList.priority1.conf",
+      "src/test/resources/conf/loadConfigFromFiles/appendIntoList.priority2.conf"
+    )
+    val expected = List(
+      98, // These three surprise me; they appear first despite coming from the lower precedence file, rather than at the end of the list.
+      99,
+      100, // 98 and 100 are included via HOCON `include`. Their order within the file is preserved.
+      8, // The next three come in order from the greater precedence file, as I'd expect.
+      9,
+      12,
+      50 // This one is slurped in via a HOCON `include`.
+    ).map(Foo(_))
```

Comment:
Ok I think I see a potential issue: in the code we are using `reduce` instead of `reduceLeft` which means that  ["The order in which operations are performed on elements is unspecified and may be nondeterministic."](http://www.scala-lang.org/api/current/index.html#scala.collection.immutable.List@reduce[A1%3E:A](op:(A1,A1)=%3EA1):A1). We should fix this ASAP.